src/commands/cmd_json.cc

```diff
@@ -204,11 +205,59 @@ class CommandJsonArrLen : public Commander {
   }
 };
 
+class CommanderJsonArrIndex : public Commander {
+ public:
+  Status Execute(Server *svr, Connection *conn, std::string *output) override {
+    redis::Json json(svr->storage, conn->GetNamespace());
+
+    std::vector<ssize_t> result;
+    int start = -1;
+    int end = std::numeric_limits<int>::max();
+
+    if (args_.size() > 4) {
+      auto parse_start = ParseInt<int>(args_[4], 10);
+      if (parse_start.IsOK()) {
+        start = parse_start.GetValue();
+      } else {
+        return {Status::RedisParseErr, errValueNotInteger};
+      }
+    }
+    if (args_.size() > 5) {
+      auto parse_end = ParseInt<int>(args_[5], 10);
+      if (parse_end.IsOK()) {
+        end = parse_end.GetValue();
+      } else {
+        return {Status::RedisParseErr, errValueNotInteger};
+      }
+    }
```

Comment:
Could you move these code to `Parse` method instead of `Execute`?

Reply:
ok, got it